src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/SessionSummaryService.kt

```diff
@@ -18,49 +26,126 @@ class SessionSummaryService(
+      SessionStatusType.ON_HOLD -> mapSessions(sessionSummaries.exemptReviews, sessionSummaries.exemptInductions)
+    }
+
+    return SessionResponses(sessions = sessions)
+  }
 
+  fun mapSessions(
+    reviews: List<ReviewSchedule>,
+    inductions: List<InductionSchedule>,
+  ): List<SessionResponse> {
+    return reviews.map {
+      SessionResponse(
+        sessionType = SessionResponse.SessionType.REVIEW,
+        prisonNumber = it.prisonNumber,
+        deadlineDate = it.reviewScheduleWindow.dateTo,
+        reference = it.reference,
+        exemptionReason = it.exemptionReason,
+        exemptionDate = if (!it.exemptionReason.isNullOrBlank()) convertInstantToLocalDate(it.createdAt) else null,
+      )
+    } + inductions.map {
+      SessionResponse(
+        sessionType = SessionResponse.SessionType.INDUCTION,
+        prisonNumber = it.prisonNumber,
+        deadlineDate = it.deadlineDate,
+        reference = it.reference,
+        exemptionReason = it.exemptionReason,
+        exemptionDate = if (!it.exemptionReason.isNullOrBlank()) convertInstantToLocalDate(it.createdAt) else null,
+      )
+    }
+  }
+
+  fun convertInstantToLocalDate(instant: Instant, zoneId: ZoneId = ZoneId.systemDefault()): LocalDate {
+    return instant.atZone(zoneId).toLocalDate()
+  }
+
+  /**
+   * Work out which schedules are due, overdue and on_hold (exempt)
+   * and then pop each one in a bucket to be counted or filtered later.
+   *
+   * Not that some sessions are disregarded - these include EXEMPT due to death or TRANSFER
+   * The complete list is configured in scheduleStatus.includeExceptionOnSummary()
+   */
+  private fun getSessionSummaries(
+    prisonerNumbers: List<String>,
+  ): SessionSummaries {
+    val sessionSummaries = SessionSummaries()
     val inductionSchedules = inductionScheduleService.getInCompleteInductionSchedules(prisonerNumbers)
     val reviewSchedules = reviewScheduleService.getInCompleteReviewSchedules(prisonerNumbers)
 
     inductionSchedules.forEach { schedule ->
       when {
-        schedule.scheduleStatus.includeExceptionOnSummary() -> exemptInductions.add(schedule.prisonNumber)
-        schedule.scheduleStatus == InductionScheduleStatus.SCHEDULED &&
-          schedule.deadlineDate < today -> overdueInductions.add(schedule.prisonNumber)
-        schedule.scheduleStatus == InductionScheduleStatus.SCHEDULED &&
-          schedule.deadlineDate > today -> dueInductions.add(schedule.prisonNumber)
+        schedule.scheduleStatus.includeExceptionOnSummary() -> sessionSummaries.exemptInductions.add(schedule)
+        schedule.scheduleStatus == INDUCTION_SCHEDULED &&
+          schedule.deadlineDate < today -> sessionSummaries.overdueInductions.add(
+          schedule,
+        )
+
+        schedule.scheduleStatus == INDUCTION_SCHEDULED &&
+          today in schedule.deadlineDate.minusMonths(2)..schedule.deadlineDate -> sessionSummaries.dueInductions.add(
```

Comment:
Induction only due two months before and up to deadline date

Reply:
Oooh, I like using a date in a range like that 👍 
I would have done isBefore/isAfter type logic, but a range is really nice 👍